bundles/org.openhab.ui/web/src/pages/settings/items/item-details.vue

```diff
@@ -184,7 +184,13 @@ export default {
       this.$store.dispatch('stopTrackingStates')
     },
     load () {
-      this.$oh.api.get(`/rest/items/${this.itemName}?metadata=semantics,${MetadataNamespaces.map((n) => n.name).join(',')}`).then((data) => {
+      this.$oh.api.get(`/rest/items/${this.itemName}?metadata=.+`).then((data) => {
+        Object.keys(data.metadata).forEach(namespace => {
+          let rs = MetadataNamespaces.find(ns => ns.name === namespace)
+          data.metadata[namespace].id = namespace
+          data.metadata[namespace].label = (rs === undefined) ? namespace : rs.label
+          data.metadata[namespace].custom = (rs === undefined)
+        })
```

Comment:
I'm not a fan of changing the response from the API just to determine whether a namespace is well-known or not (and add fields with the label and a flag).
This logic must be enclosed in the item-metadata-menu.vue component where it doesn't depend on the parent component already having performed that logic before passing the data in the prop.

Reply:
I tried this but I ended up with an v-if statement in the v-for which is not allowed to be used (the linter told me), so I came up with this idea.

Also it is not only to detect if it is a custom namespace but also to enrich the API information with label that is known from the MetadataNamespaces and for those that are unknown, detect that it is a custom namespace. Also I need to detect that it is a custom namespace. 

The alternative I could think of could the be that the whole label enrichment would then be made in the computed section instead which to me seems to be more complicated than my approach and would require a completely different approach but I may be the wrong person then to do that, fair enough... maybe someone else takes over then.

Reply:
> The alternative I could think of could the be that the whole label enrichment would then be made in the computed section instead which to me seems to be more complicated than my approach and would require a completely different approach but I may be the wrong person then to do that, fair enough... maybe someone else takes over then.

No worries. It's indeed the more correct implementation as this logic is only for the display purposes of the item-metadata-menu component, so processing the data is this component's concern only, and you can simply pass it the item unmutated (mitigates mistakes like the missing change in model.vue as the component is used there too).

See https://github.com/openhab/openhab-webui/pull/1623/commits/e85af092bbae1b44d24c19b4d0bf5fe28ac2db95 for the implementation.
I fixed the model page and made some cleanups.